Orange/canvas/report/owreport.py

```diff
@@ -301,9 +328,16 @@ def save_report(self):
         elif extension == ".report":
             self.save(filename)
         else:
-            frame = self.report_view.page().currentFrame()
-            with open(filename, "w", encoding="utf-8") as f:
-                f.write(frame.documentElement().toInnerXml())
+            page = self.report_view.page()
+
+            def save_html(contents):
+                with open(filename, "w", encoding="utf-8") as f:
+                    f.write(contents)
+
+            if webview.HAVE_WEBENGINE:
+                page.toHtml(save_html)
+            else:
+                save_html(page.mainFrame().toHtml())
```

Comment:
Isn't a minimal compatible wrapping interface at one single place better than "ifdefs"  all around the codebase?